### PR TITLE
Bug 2060553: Separate the allPodsSelected into egress and ingress

### PR DESCRIPTION
**Pull request by knobunc**
The code did not differentiate whether all pods were selected for egress or for ingress.  This changes to track the two separately.

### CODE REVIEW

File: pkg/network/node/networkpolicy.go [hunk 1]
```diff
@@ -357,18 +357,23 @@ func (np *networkPolicyPlugin) generateNamespaceFlows(otx ovs.Transaction, npns
 		return
 	}
 
-	var allPodsSelected, affectsIngress, affectsEgress bool
+	var allIngressPodsSelected, allEgressPodsSelected, affectsIngress, affectsEgress bool
```
Comment (JacobTanenbaum) [inline pkg/network/node/networkpolicy.go:360]: nit: could you change the variables to `allPodsSelectedIngress` and `allPodsSelectedEgress`

**Review by danwinship — CHANGES_REQUESTED** [review]
I started to rewrite this to refactor the code better, but since we're doing a last-minute backport, let's stick with the less-rewriting approach for now and we can clean it up later.

**Comment by wking** [comment]
Using a delimiter that the Bugzilla Prow plugin will recognize:

/retitle Bug 2060553: Separate the allPodsSelected into egress and ingress

**Comment by JacobTanenbaum** [comment]
lgtm

**Comment by mffiedler** [comment]
/label qe-approved

Verified on cluster-bot cluster built from this PR using the reproducer in https://bugzilla.redhat.com/show_bug.cgi?id=2060553#c13.   Service domain resolvable with the patch.

**Comment by knobunc** [comment]
/overrride ci/prow/e2e-aws

Per the flake data on "should support sysctls" it seems to be unrelated.  All other tests pass.

**Comment by knobunc** [comment]
/override ci/prow/e2e-aws

Now with less pirate.

**Pull request merged**